### DISCUSSION TITLE
Scene Restructure work

### DISCUSSION THREAD

**Discussion by jjburton** [jjburton/cgmTools#220] [Ideas]
Testing branch: features/sceneStructure

This is a big one. Wanting to retool how subType and variant work as well as allow more flexible workflow.

-first level of folders are subTypes and fill the top menu for the first column
- second 

### Structure

```
-AssettType
---AssetName
------subType (dirs)
---------sub1(sub)
------------sub2 (variant)
```

Single
```
- Category
---files 
```

Simple Asset
```
- Category
---Asset
------files 
```

Subtype (no sub, no var)
```
- Category
---Asset
------subType
---------files
```

Sub1 (hasSub)
```
- Category
---Asset
------subType 
---------sub1 (sub)
------------files
```

Sub2 (hasVariant)
```
- Category
---Asset
------subType
---------sub1 (sub)
------------sub2 (variant)
---------------files
```

**Comment by jjburton** [comment]
### Category list

- [x] see non registered categories
- [ ] any directory works

- Added framework for this to work.

**Comment by jjburton** [comment]
### asset list
- [ ]

**Comment by benngarnish** [comment]
I wonder if the tooling would allow you to source a new top level folder structure and from what it reads to give you the option to tag the relative folders in a way that can update the mrsProject config file. 
what I’m thinking here is that the top level folder would be selected and if the config file does t match the folder variation types then it pops up with a UI to tag the folders that it doesn’t know and update the config file so that when sending to mrsScene the system works even if you then wanted to start adding more assets. 

This would be particularly useful when working in other studio environments, that have a very different templating system. 

Something else would be multiple directories. 
Does a studio have multiple directories for where assets live vs shots?

The same goes for the export directories. Is there a way of having multiple locations?


Asset working files go here: ....
Asset export files go here: ....

Shots working files go here: ....
Shots export files go here: ....


B.

**Reply by jjburton** [reply]
My current thinking is to just treat sub folders as levels, it if finds a directory that asset has subs to walk.

On the export, yeah that's possible but a bit out of scope of this push I'm on. Shouldn't be a big thing once I'm done with this. Could be something for you to tackle ;)

**Comment by jjburton** [comment]
### Week of 4.12

- [x] rework hasSub
- [x] rework hasVariant
- [x] mask meta dir and add dir mask to handle
- [ ] selected subtype dir shouldn't try to load variations if file
- [ ] selected variation shouldn't try to load if file

**Reply by jjburton** [reply]
### Paths
Write some calls to get consistently get paths

- [x] category
- [x] subtype
- [x] variation
- [x] version
- [x] file

**Comment by jjburton** [comment]
### Export/Batch issues

- reworked storage of data to account for new euler/tangent options
- rework how batch maya py file paths are stored
- change how export file name is built and more

**Comment by jjburton** [comment]
### Rig Ref maker
- added to version list
- create SubTypeRef

**Comment by jjburton** [comment]
### Batch export working...
- added new calls for batching rig,export,cutscene
- added new pop menu to version/subtype list for this
-

**Comment by jjburton** [comment]
### MA/MB
- added to top menu
- version save now honors
- added filters to save here calls

**Comment by jjburton** [comment]
### Export Select/Geo
- fix to prep and bake to not assume name space to better get a export set
-

**Comment by jjburton** [comment]
### 05.21
- OSUTILS - New call | mkdir_recursive || call for verifying a folder path structure
- SCENE/backAndPrep | working through basic scene export, maya file pref option pass initial pass,

**Comment by jjburton** [comment]
### Asset Rename
This isn't working, need to change.
- [x] Pop up to get name
- [x] Find/replace for each level of the walk

**Comment by jjburton** [comment]
### Week of 05.24
- Asset rename added
- Bug | ref file now uses asset name
